src/components/replacementCard/ReplacementCard.js

```diff
@@ -1,14 +1,115 @@
 import Paper from '@material-ui/core/Paper';
 import Grid from '@material-ui/core/Grid';
 import ReplacementCardStyles from "./ReplacementCardStyles";
+import TextField from "@material-ui/core/TextField";
+import {useContext, useEffect, useState} from "react";
+import Button from "@material-ui/core/Button";
+import List from '@material-ui/core/List';
+import Divider from '@material-ui/core/Divider';
+import ListItemText from '@material-ui/core/ListItemText';
+import ListItem from '@material-ui/core/ListItem';
+import CustomerService from "../../shared/services/CustomerService";
+import MemberService from "../../shared/services/MemberService";
+import {GlobalContext} from "../../context/GlobalState";
+import moment from "moment";
+
+const ReplacementCard = (props) => {
+    const {customerData, addTransactions, addCoupons, deleteCoupons, ...rest} = useContext(GlobalContext);
+    const [replacementCard, setReplacementCard] = useState('');
+    const [existingCards, setExistingCards] = useState([]);
+    const [customer, setCustomer] = useState( []);
+
+    useEffect(() => {
+       getAllCards();
+    },[]);
+
+    const getActivationDate = (card) =>{
+        if(customer && customer.customerCards){
+        const cardFound = customer.customerCards.find(item => item.ciid === card.ciid)
+        return cardFound && moment(cardFound.activationDate).format("DD-MM-yyyy");
+        }
+    }
 
-const ReplacementCard = () => {
     const classes = ReplacementCardStyles();
+
+   const replacementCardHandler=(event) => {
+        setReplacementCard(event.target.value)
+    }
+    const addReplacementCard =() => {
+       const {salesDivision, subsidiary} = props;
+     const dataToBeSaved = {
+         ...customer,
+         ciid:replacementCard,
+         salesDivision,
+         subsidiary
+     }
+        CustomerService.validateCardNumber(dataToBeSaved)
+            .then(() => {
+                const data= CustomerService.createCustomerUpsertData(customer, props.configData)
+                data.cardCiid = dataToBeSaved.ciid;
+                CustomerService.upsertCustomer(data)
+                    .then((result) => {
+                   console.log(result);
+                   getAllCards();
+                    })
+            })
+    }
+
+    const getAllCards = () => {
+        const {salesDivision, subsidiary, partyUid,activationDate} = props;
+        const data = {
+            salesDivision,
+            subsidiary,
+            partyUid,
+            activationDate
+        }
+        MemberService.getHousehold(data)
+            .then( (result) => {
+                console.log('result',result)
+                setExistingCards(result.data[0].householdCards)
+            })
+        CustomerService.selectCustomer(data)
```

Comment:
customer information data are in the customerData object. why do you make a new request?